### PR TITLE
http: unifying client types

### PR DESCRIPTION
**Pull request by alyssawilk**
It has been a pet peeve of mine for years that the test upstream and test client have different enums for the same codecs.
Unifying them, but leaving legacy typedefs for folks who don't want to update all their filters/tests.

Risk Level: Low (enum change, theoretical legacy support)
Testing: n/a
Docs Changes: n/a
Release Notes: n/a

### CODE REVIEW

File: test/integration/fake_upstream.h [hunk 1]
```diff
@@ -413,22 +413,24 @@ class FakeConnectionBase : public Logger::Loggable<Logger::Id::testing> {
  */
 class FakeHttpConnection : public Http::ServerConnectionCallbacks, public FakeConnectionBase {
 public:
-  enum class Type { HTTP1, HTTP2, HTTP3 };
-  static absl::string_view typeToString(Type type) {
+  // This is a legacy alias.
+  using Type = Envoy::Http::CodecType;
+  // using Type = Envoy::Http::CodecType;
```
Comment (snowp) [inline test/integration/fake_upstream.h:418]: Remove this line?

**Review by snowp — CHANGES_REQUESTED** [review]
Nice, thanks! One small comment

**Review by snowp — DISMISSED** [review]
Thanks!

**Comment by alyssawilk** [comment]
I think this PR is just too large for clang tidy to finish.  @snowp would you be Ok if I just force-merge over that?

**Comment by snowp** [comment]
Fine with me as long as we don't break main as a result :)

**Review by snowp — APPROVED** [review]

**Comment by alyssawilk** [comment]
yeah, I think it's just timing out on the diffs.  will revert if I'm wrong :-)

**Pull request merged**